src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -0,0 +1,591 @@
+import { Pagination, PaginationProps, Tooltip } from 'antd';
+import { useEffect, useRef, useState } from 'react';
+
+import { useArweaveCompositeProvider } from '../../../../hooks';
+import { useGlobalState } from '../../../../state/contexts/GlobalState';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  VALIDATION_INPUT_TYPES,
+} from '../../../../types';
+import { isArweaveTransactionID } from '../../../../utils';
+import eventEmitter from '../../../../utils/events';
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  CircleCheck,
+  CirclePlus,
+  CircleXIcon,
+} from '../../../icons';
+import { Loader } from '../../../layout';
+import ValidationInput from '../ValidationInput/ValidationInput';
+
+function NameTokenSelector({
+  walletAddress,
+  selectedTokenCallback,
+}: {
+  walletAddress: ArweaveTransactionID;
+  selectedTokenCallback: (id: ArweaveTransactionID | undefined) => void;
+}) {
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const [{ pdnsSourceContract }] = useGlobalState();
+
+  const [searchText, setSearchText] = useState<string>();
+  const [tokens, setTokens] = useState<{
+    [x: string]: Partial<PDNTContractJSON>;
+  }>();
+  const [loading, setLoading] = useState(false);
+  const [filteredTokens, setFilteredTokens] =
+    useState<
+      Array<{ id: string; name?: string; ticker?: string } | undefined>
+    >();
+  const [selectedToken, setSelectedToken] = useState<
+    { id: string; name: string; ticker: string } | undefined
```

Comment:
see comment about `Pick` vs. `Omit` and standardize `tokens` and `selectedToken` to have matching types